lib/src/modules/conditional.dart

```diff
@@ -890,6 +889,43 @@ enum ConditionalType {
   priority
 }
 
+/// Shorthand for a [Conditional] inside a [Case] block.
+/// Returns a [Logic] signal based on the expression and conditions.
+/// The width of the result is determined by the width of the value of the
+/// conditions.
+Logic cases(Logic expression, Map<Logic, dynamic> conditions, {int? width}) {
```

Comment:
Should we support inferring widths for the keys of `conditions` also?  E.g. then you could put `LogicValue`, `Logic`, `int`, etc. as the match for the `expression`.